### DISCUSSION TITLE
Group remaining @types PR's

### DISCUSSION THREAD

**Discussion by ceyhun** [renovatebot/renovate#22141] [Suggest an Idea]
#### This is a:

* [X] Feature request

#### I'm using:

* [X] The Renovate GitHub App

#### Please describe the issue:

I like that `@types/nock` and `nock` are grouped automatically.
But I also want to group all the other `@types` that remain.
When I use `group:definitelyTyped`, `@types/nock` and `nock` becomes seperated.

See also: https://github.com/renovateapp/config-help/issues/25

**Comment by rarkins** [comment]
Example with nock:

If `nock` also has an upgrade then the types should be in `renovate/nock-1.x` otherwise it should be in the definitely types grouping. 

Made more complex because the definitely types group is a regular package rule and not something internal to Renovate.

Also presumably there are people who’d like the option to not dynamically group with `nock`

**Comment by rarkins** [comment]
I’m starting to think we may need to make all special types handling to be native feature and not via package rule, including:
- implicit grouping with its parent (eg `@types/nock` with `nock`)
- fallback grouping with other `@types`
- ensuring the types major.minor does not exceed its parent’s major.minor

**Comment by ghost** [comment]
We have the following configuration rule set:

```json
{
  "extends": [
    "monorepo:angular1"
  ],
  "allowedVersions": "<=1.5",
  "minor": {
    "automerge": false
  }
}
```

I recently received a pull request for: `chore(deps): update dependency @types/angular to ~1.6.0`

It was unexpected since I expected our types packages to follow the same `allowedVersions` rule.

The immediate issue is that `monorepo:angular1` does not include related `@types` packages.

Neither does my custom configuration shown above.

A quick fix would be to update my own configuration, then, perhaps, upstream something to `renovate-config`. In the meantime, I wanted to bring it up here since there's an on-going discussion about handling `@types` packages.

**Comment by rarkins** [comment]
I'm going to close this unless we have a feasible proposal for how to handle it
